lib/reports/ReportBaseCrash.js

```diff
@@ -19,58 +21,35 @@ class ReportBaseCrash extends ReportBase {
   }
 
   /**
-   * Parses an ID in the format `{centrelineType}/{centrelineId}`, and returns it
-   * as a location from {@link CentrelineDAO} for further processing.
+   * Parses a `CompositeId`, and returns it as an array of centreline locations from
+   * {@link CentrelineDAO} for further processing.
    *
-   * @param {string} rawId - ID to parse
+   * @param {string} s1 - `CompositeId` to parse
    * @throws {InvalidReportIdError}
    */
-  async parseId(rawId) {
-    const parts = rawId.split('/');
-    if (parts.length !== 2) {
-      throw new InvalidReportIdError(rawId);
-    }
-
-    let centrelineType = parts.shift();
-    centrelineType = parseInt(centrelineType, 10);
-    if (Number.isNaN(centrelineType)) {
-      throw new InvalidReportIdError(rawId);
-    }
-
-    let centrelineId = parts.shift();
-    centrelineId = parseInt(centrelineId, 10);
-    if (Number.isNaN(centrelineId)) {
-      throw new InvalidReportIdError(rawId);
-    }
-
+  async parseId(s1) {
     try {
-      const location = await CentrelineDAO.byIdAndType(centrelineId, centrelineType);
-      if (location === null) {
-        throw new InvalidReportIdError(rawId);
-      }
-      return location;
+      const features = CompositeId.decode(s1);
+      const locations = await CentrelineDAO.byIdsAndTypes(features);
+      return Array.from(locations.values());
     } catch (err) {
       if (err instanceof InvalidCentrelineTypeError) {
-        throw new InvalidReportIdError(rawId);
+        throw new InvalidReportIdError(s1);
```

Comment:
don't rethrow this as `InvalidReportIdError` - if the decoding works, it should guarantee that the resulting features have valid `centrelineType` values.